### ISSUE TITLE
Three Column ISO Document

### ISSUE THREAD

**Issue by jasonnicholson** [metanorma/metanorma#342]
I am the covenor related to updating ISO 5598 which is a three column document. Do you know if Metanorma can support such a document? How would you recommend managinga  three language document with Metanorma?

[ISO 5598 on the Online Browsing Platform](https://www.iso.org/obp/ui#iso:std:iso:5598:ed-3:v1:en)
![image](https://github.com/metanorma/mn-samples-iso/assets/1058191/8f620899-78f5-448e-a74a-7166a38848d7)

The working group likes seeing the definitions side by side:
![image](https://github.com/metanorma/mn-samples-iso/assets/1058191/9e4c042b-c6fe-4118-b38a-eaf07a13a520)

**Comment by ronaldtse** [comment]
@jasonnicholson thank you for raising the question!

This is interesting as there are multiple questions:
- What is the best way to manage terminology when publishing in Metanorma?
- Does Metanorma support publishing a trilingual document in a 3-column layout?
- Does ISO support publishing in a 3-column layout?

Question 1. The best way to manage a multi-lingual terminology database is through Glossarist/Paneron, which its dataset can directly be processed by Metanorma. 

Question 2. We currently have a PR that aims to support a bilingual / trilingual multi-column layout. Our problem is that we do not have a current enough trilingual standard to work with. We would love to help your work here in finalizing support for such a layout. 

Question 3. Metanorma produces both PDF and STS output. The latter is used by ISO/CS to either import into the OSD or to run through their publishing toolchain. However, the production capabilities of ISO/CS are rather limited, and I suspect that their STS capabilities will not support a trilingual 3-column layout. In our experience, for special PDFs that ISO/CS cannot produce, they will directly edit the Metanorma produced PDF and wrap with their own cover pages. So publishing the PDF shouldn’t be a problem.

**Comment by jasonnicholson** [comment]
Thank you for the answers.

>Question 1. The best way to manage a multi-lingual terminology database is through Glossarist/Paneron, which its dataset can directly be processed by Metanorma.

We will look into Glossarist/Paneron.

>Question 2. We currently have a PR that aims to support a bilingual / trilingual multi-column layout. Our problem is that we do not have a current enough trilingual standard to work with. We would love to help your work here in finalizing support for such a layout.

Very good. :) We may be able to help here. I need to discuss this with ISO CS and the WG. Give me some time. I want to provide you with a document to test out. We have a trilingual document.


>Question 3. Metanorma produces both PDF and STS output. The latter is used by ISO/CS to either import into the OSD or to run through their publishing toolchain. However, the production capabilities of ISO/CS are rather limited, and I suspect that their STS capabilities will not support a trilingual 3-column layout. In our experience, for special PDFs that ISO/CS cannot produce, they will directly edit the Metanorma produced PDF and wrap with their own cover pages. So publishing the PDF shouldn’t be a problem.

ISO published 5598:219 as a trilingual document. I don't know the details. I am working on understanding the ISO toolchain to minimize the work on the WG editors and ISO CS editors. Do you happen to know a good contact at ISO to talk about the toolchain? So far, ISO CS has pointed me to OSD user documentation which wasn't helpful.

**Comment by ronaldtse** [comment]
In our process of encoding ISO 5598:2019 into Metanorma, I noticed that the French version already contains a mistake in the identification of the TC (should be TC 131, French version says TC 133):

<img width="931" alt="image" src="https://github.com/metanorma/metanorma/assets/11865/5614a4d6-4d2a-42ca-a6bb-73e22a037151">

**Comment by ronaldtse** [comment]
This document is also interesting as it has:
* terms using the German language

* terms with 2 domains
```adoc
[[term_3.2.61]]
==== axial piston pump
alt:[angled piston pump]
domain:[bent axis design]
domain:[hydraulic]
```

* Figures shared across multiple terms

```adoc
[[term_3.2.384]]
==== idling pressure
...
NOTE: See Figure 20.

[[term_3.2.441]]
==== maximum pressure
...
NOTE: See Figure 20.
...
```

* Figures with key

```adoc
Figure 20 — Illustration of pressure terms related to fluid power systems

Key
X	time	9	pressure surge
Y	pressure	10	pressure fluctuations
1	absolute pressure	11	idling pressure
2	atmospheric pressure	12	pressure drop
3	negative gauge pressure	13	minimum working pressure
4	positive gauge pressure	14	working pressure range
5	steady-state pressure	15	maximum working pressure
6	pressure pulsation	16	maximum pressure
7	pressure pulse	17	operating pressure range
8	pressure peak
```

* Extensive grammar info on designations (on alt, on deprecated)
```adoc
[[term_3.2.754]]
==== T-​Verschraubung,f
alt:[T-​Verbindungsstutzen,m]
```

```adoc
[[term_3.2.777]]
==== Zweistufenfilter,m,n
```

```adoc
[[term_3.1.1]]
==== Ist-; tatsächlich,Adj.
```

```adoc
[[term_3.1.5]]
==== wirksam; effektiv,Adj.
```

```adoc
[[term_3.1.7]]
==== Bemessungs-; bemessen,Adj.
```

* Domains that are identified with concept mentions (the domain is a cross-referenced term)

```adoc
[[term_3.2.413]]
==== Überdeckung,f
domain:[Stetigventil (3.2.136)]
```

* Designations that belong to different domains:
```adoc
[[term_3.2.633]]
==== Drei-​Wege-​Druckminderventil,n <Hydraulik>
alt:[Druckminderventil mit Rückstrommechanismus,n <Pneumatik>]
```

```adoc
[[term_3.2.776]]
==== Zwei-​Wege-​Ventil,n〈Hydraulik〉
alt:[Zwei-Anschlüsse-Ventil,n〈Pneumatik〉]
```

* Usage info

```adoc
[[term_3.2.788]]
==== Ejektor,m
alt:[Strahlpumpe (mit Absaugvorrichtung), f]
domain:[Pneumatik]
```

* Concept mentions on word prefixes
```adoc
[[term_3.2.808]]
==== Flüssigkeitsschlag,m
domain:[Hydraulik]

oszillierendes Druck-(3.2.560)-Pulsphänomen, das durch eine schnelle Abnahme des Volumenstroms (3.2.303) erzeugt wird, der gegen einen kompressiblen Energiespeicher wirkt
```

**Comment by jasonnicholson** [comment]
> In our process of encoding ISO 5598:2019 into Metanorma, ...

How did you get a copy of the ISO 5598:2019 document? Right now, I am waiting on ISO CS and my secretary to get back to myself and the project leader related to this document.

>... I noticed that the French version already contains a mistake in the identification of the TC (should be TC 131, French version says TC 133):

Thank you. We will get that fixed in the next version.

> [[term_3.2.61]]
> ==== axial piston pump
> alt:[angled piston pump]
> domain:[bent axis design]
> domain:[hydraulic]

I am new to vocabulary. I was not aware of the domain. ISO directives part 2 does not have an example of a domain and there is only one reference to the word domain in the directives part 2.


With regard to the rest of your statements, there interesting observations. Is it only interesting or do you have a recommendation?

**Comment by ronaldtse** [comment]
@jasonnicholson there are actually a few ways to get the document:
* Automated conversion
    * Ask ISO/CS to provide the STS XML document. Metanorma's `mnconvert` tool allows you to convert STS XML into Metanorma AsciiDoc, which is the textual markup format used by Metanorma.
* Manual conversion
    * Ask ISO/CS to provide the original Word document
    * Obtain the content from the ISO OBP (this is what I did) -- informative content is freely available
    * Transcribe content from PDF files

We have actually converted the document for demonstration purposes (except for the concept mentions, which will be dealt easier if we have the STS XML, or we can parse the OBP HTML into an STS-like format):

* https://github.com/metanorma/iso-5598

This is a private repository right now and have shared it with you, you should have received an invite by now.

> I am new to vocabulary. I was not aware of the domain. ISO directives part 2 does not have an example of a domain and there is only one reference to the word domain in the directives part 2.

Yes this is the only example in ISO DIR 2. It is currently possible to have a term in two different domains in Metanorma, so it works. 

<img width="721" alt="Screenshot 2023-11-29 at 6 24 26 PM" src="https://github.com/metanorma/metanorma/assets/11865/33b69dc3-7bff-4ff1-8b62-a00f19e29f5f">


> With regard to the rest of your statements, there interesting observations. Is it only interesting or do you have a recommendation?

Most of these related to features we need to implement in Metanorma, and we've filed issues for them in the shared repo now.

With regards to "designations in different domains", I'm not sure if it's even supported by ISO 10241-1. I'll have to double check.

**Comment by ronaldtse** [comment]
Just as a teaser, this is how it looks like in French now.

<img width="566" alt="Screenshot 2023-11-29 at 6 25 56 PM" src="https://github.com/metanorma/metanorma/assets/11865/d73bc840-4b57-45b3-98db-668350517ab1">

**Comment by jasonnicholson** [comment]
Hi Ronald. Can you send the invitation to join Metanorma again? I tried to use it today and could not.

I apologize. Last week was busy and over the weekend I finished my relocation to North Carolina.

**Comment by ronaldtse** [comment]
@jasonnicholson I've sent the invitation again. Sorry for the late send!

**Comment by jasonnicholson** [comment]
Thank you. I accepted the request. I hope to do some experimenting between Christmas and new years.

**Comment by ronaldtse** [comment]
@jasonnicholson any update to the project so far? We have implemented "Figures with key" a while ago so another step closer.

**Comment by jasonnicholson** [comment]
This has been on hold for the last few months. I don't have anything to report right now.

**Comment by jasonnicholson** [comment]
ISO 5598 is going forward. 

I did try to get the XML, but multiple attempts have yielded zero replies from the important people. I will not be able to obtain the XML.

I have a Word format, which you scraped from the HTML.

I plan to take some time this summer to look at this.
